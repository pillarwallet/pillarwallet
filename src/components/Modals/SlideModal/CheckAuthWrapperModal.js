// @flow
/*
    Pillar Wallet: the personal data locker
    Copyright (C) 2019 Stiftung Pillar Project

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
*/

// CheckAuthWrapperModal is a copy of the old version of <SlideModal>
// (controlled by isVisible prop) with unused and unchanging props removed.
//
// (old) <SlideMedal> props with constant values were:
// {
//   centerTitle: true,
//   showHeader: true,
//   fullScreen: true,
// }
//
// Can be deleted after <CheckAuth> is refactored to use the new modal mechanism.

import * as React from 'react';
import Modal from 'react-native-modal';
import styled, { withTheme } from 'styled-components/native';
import Root from 'components/Root';
import Toast from 'components/Toast';
import { Wrapper } from 'components/Layout';
import HeaderBlock from 'components/HeaderBlock';
import { Keyboard } from 'react-native';
import { getThemeColors, themedColors } from 'utils/themes';
import type { Theme } from 'models/Theme';

export type ScrollToProps = {
  x?: number,
  y: number,
  animated: boolean,
}

type Props = {|
  title?: string,
  children?: React.Node,
  onModalHide?: Function,
  isVisible: boolean,
  theme: Theme,
|};

const ContentWrapper = styled.View`
  width: 100%;
  height: 100%;
  padding-top: 20px;
  background-color: ${themedColors.surface};
`;

const ModalContent = styled.View`
  flex-direction: column;
  flex: 1;
  padding: 0;
`;

const ModalOverflow = styled.View`
  width: 100%;
  height: 100px;
  margin-bottom: -100px;
  background-color: ${themedColors.card};
`;

class CheckAuthWrapperModal extends React.Component<Props, *> {
  constructor(props) {
    super(props);
    this.state = {
      contentHeight: 0,
    };
  }

  hideModal = () => {
    Keyboard.dismiss();
    const TIMEOUT = Toast.isVisible() ? 150 : 0;
    if (Toast.isVisible()) {
      Toast.closeAll();
    }
    const timer = setTimeout(() => {
      if (this.props.onModalHide) {
        this.props.onModalHide();
      }
      clearTimeout(timer);
    }, TIMEOUT);
  };

  onModalBoxLayout = (event) => {
    const height = event.nativeEvent?.layout?.height || 0;
    if (this.state.contentHeight !== height) {
      this.setState({
        contentHeight: height,
      });
    }
  }

  render() {
    const {
      children,
      title,
      isVisible,
      theme,
    } = this.props;

    const colors = getThemeColors(theme);

    const modalInner = (
      <React.Fragment>
        <HeaderBlock
          leftItems={[]}
          centerItems={[{ title }]}
          rightItems={[{ close: true }]}
          noBottomBorder
          noPaddingTop
          onClose={this.hideModal}
          wrapperStyle={{ backgroundColor: 'transparent' }}
          noHorizonatalPadding={false}
          noBack
          forceInsetTop="never" // eslint-disable-line i18next/no-literal-string
        />
        <ModalContent>
          {children}
        </ModalContent>
        <ModalOverflow />
      </React.Fragment>
    );

    const animationTiming = 400;
    return (
      <Modal
        isVisible={isVisible}
        onSwipeComplete={this.hideModal}
        onBackdropPress={this.hideModal}
        backdropOpacity={1}
        backdropColor={colors.surface}
        onBackButtonPress={this.hideModal}
        animationInTiming={animationTiming}
        animationOutTiming={animationTiming}
        animationIn="slideInUp"
        animationOut="slideOutDown"
        swipeDirection="down"
        style={{
          margin: 0,
          position: 'relative',
          zIndex: 10,
        }}
      >
        <Root>
          <ContentWrapper>
            <Wrapper onLayout={this.onModalBoxLayout} fullScreen>
              {modalInner}
            </Wrapper>
          </ContentWrapper>
        </Root>
      </Modal>
    );
  }
}

export default (withTheme(CheckAuthWrapperModal): React.AbstractComponent<$Diff<Props, { theme: Theme }>>);
