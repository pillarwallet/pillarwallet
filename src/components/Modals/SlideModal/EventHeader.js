// @flow
import * as React from 'react';
import { Platform } from 'react-native';
import { BoldText, BaseText } from 'components/Typography';
import { fontTrackings, baseColors, fontSizes, spacing } from 'utils/variables';
import styled from 'styled-components/native';
import IconButton from 'components/IconButton';
import Icon from 'components/Icon';
import {
  TRANSACTION_SENT,
  TRANSACTION_SENT_PENDING,
  TRANSACTION_RECEIVED,
  TRANSACTION_RECEIVED_PENDING,
  CONNECTION_INCOMING,
  CONNECTION_SENT,
  CONNECTION_MADE,
} from 'constants/eventsConstants';

type Props = {
  onBack?: Function,
  onClose?: Function,
  onCloseText?: string,
  onNextPress?: Function,
  nextText?: string,
  nextIcon?: string,
  title?: string,
  centerTitle?: boolean,
  noPadding?: boolean,
  noMargin?: boolean,
  flexStart?: boolean,
  light?: boolean,
  style?: Object,
  headerRightFlex?: string,
  overlay?: boolean,
  eventType?: string,
}

const getEventInfo = (eventType) => {
  switch (eventType) {
    case TRANSACTION_SENT:
      return {
        title: 'Success',
        background: baseColors.freshEucalyptus,
        subtitle: 'OCTOBER 31, 2017 21:14',
        iconName: 'tick-circle',
      };
    case TRANSACTION_RECEIVED:
      return {
        title: 'Success',
        background: baseColors.freshEucalyptus,
        subtitle: 'OCTOBER 31, 2017 21:14',
        iconName: 'tick-circle',
      };
    case TRANSACTION_SENT_PENDING:
      return {
        title: 'Pending',
        background: baseColors.burningFire,
        subtitle: '4 MIN 19 SEC AGO',
        iconName: 'pending-circle',
      };
    case TRANSACTION_RECEIVED_PENDING:
      return {
        title: 'Pending',
        background: baseColors.burningFire,
        subtitle: '4 MIN 19 SEC AGO',
        iconName: 'pending-circle',
      };
    case CONNECTION_INCOMING:
      return {
        title: 'Incoming connection',
        background: baseColors.rose,
        subtitle: '20 APRIL, 16:30',
        iconName: 'connection-circle',
      };
    case CONNECTION_SENT:
      return {
        title: 'Request sent',
        background: baseColors.electricBlue,
        subtitle: '25 APRIL, 18:05',
        iconName: 'connection-circle',
      };
    case CONNECTION_MADE:
      return {
        title: 'Connection established',
        background: baseColors.cerulean,
        subtitle: '20 APRIL, 16:31',
        iconName: 'connection-circle',
      };
    default:
      return {
        title: 'Event',
        background: baseColors.darkGray,
        subtitle: '20 APRIL, 16:31',
        iconName: 'info-circle',
      };
  }
};


const Wrapper = styled.View`
  width: 100%;
  border-top-left-radius: 30px;
  border-top-right-radius: 30px;
  overflow: hidden;
  padding: ${spacing.mediumLarge}px 50px;
  background-color: ${props => props.background};
  align-items: center;
  justify-content: center;
`;

const CloseIcon = styled(IconButton)`
  position: absolute;
  height: 24px;
  width: 24px;
  top: ${Platform.select({
    ios: '20px',
    android: '19px',
  })}
  right: 12px;
  opacity: 0.7;
`;

const EventTitle = styled(BoldText)`
  font-size: ${fontSizes.mediumLarge}px;
  letter-spacing: ${fontTrackings.tiny}px;
  color: ${baseColors.white};
  margin: 2px 0;
  text-align: center;
`;

const EventSubtitle = styled(BaseText)`
  font-size: ${fontSizes.tiny}px;
  letter-spacing: ${fontTrackings.mediumLarge}px;
  color: ${baseColors.white};
  margin: 2px 0;
  text-align: center;
`;

const EventIcon = styled(Icon)`
  opacity: 0.7;
  margin-top: ${spacing.medium}px;
`;

const EventHeader = (props: Props) => {
  const {
    onClose,
    eventType,
  } = props;

  const thisEvent = getEventInfo(eventType);

  return (
    <Wrapper background={thisEvent.background}>
      <CloseIcon
        icon="close"
        color={baseColors.white}
        onPress={onClose}
        fontSize={fontSizes.small}
      />
      <EventTitle>{thisEvent.title}</EventTitle>
      <EventSubtitle>{thisEvent.subtitle}</EventSubtitle>
      <EventIcon
        name={thisEvent.iconName}
        style={{
          color: baseColors.white,
          fontSize: 58,
        }}
      />
    </Wrapper>
  );
};

export default EventHeader;
