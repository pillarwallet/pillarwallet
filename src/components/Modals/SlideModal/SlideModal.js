// @flow
/*
    Pillar Wallet: the personal data locker
    Copyright (C) 2019 Stiftung Pillar Project

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
*/
import * as React from 'react';
import type { Node as ReactNode } from 'react';
import styled, { withTheme } from 'styled-components/native';
import type { LayoutEvent } from 'react-native/Libraries/Types/CoreEventTypes';
import isEmpty from 'lodash.isempty';
import pick from 'lodash.pick';

import Modal from 'components/Modal';
import { Wrapper } from 'components/Layout';
import HeaderBlock from 'components/HeaderBlock';

import { spacing } from 'utils/variables';
import { getThemeColors } from 'utils/themes';

import type { ScrollToProps } from 'components/Modal';
import type { Theme } from 'models/Theme';
import type { OwnProps as HeaderProps } from 'components/HeaderBlock';

export type { ScrollToProps } from 'components/Modal';

type ModalProps = {|
  onModalWillHide?: () => void,
  onModalHide?: () => void,
  onModalShow?: () => void,
  avoidKeyboard?: boolean,
  scrollOffset?: number,
  scrollOffsetMax?: number,
  scrollTo?: (_: ScrollToProps) => void,
|}

type OwnProps = {|
  ...ModalProps,
  title?: string,
  children?: ReactNode,
  noClose?: boolean,
  fullScreen?: boolean,
  showHeader?: boolean,
  hideHeader?: boolean,
  centerTitle?: boolean,
  centerFloatingItem?: ReactNode,
  backgroundColor?: string,
  eventDetail?: boolean,
  noSwipeToDismiss?: boolean,
  noPadding?: boolean,
  headerLeftItems?: $PropertyType<HeaderProps, 'leftItems'>,
  sideMargins?: number,
  noTopPadding?: boolean,
  headerProps?: HeaderProps,
  insetTop?: boolean,
|};

type Props = {|
  ...OwnProps,
  theme: Theme,
|};

type State = {|
  contentHeight: number,
|};

const themes = {
  default: {
    padding: `0 ${spacing.layoutSides}px`,
    borderRadius: '30px',
  },
  fullScreen: {
    padding: 0,
    borderRadius: 0,
  },
  eventDetail: {
    padding: 0,
    borderRadius: '30px',
  },
  noPadding: {
    padding: 0,
    borderRadius: '30px',
  },
};

const getTheme = (props: Props) => {
  if (props.fullScreen) {
    return themes.fullScreen;
  }
  if (props.eventDetail) {
    return themes.eventDetail;
  }
  if (props.noPadding) {
    return themes.noPadding;
  }
  return themes.default;
};

const ContentWrapper = styled.View`
  width: 100%;
  height: 100%;
  ${props => props.fullScreen && !props.noTopPadding ? 'padding-top: 20px;' : ''}
  ${props => props.bgColor && props.fullScreen ? `background-color: ${props.bgColor};` : ''}
`;

const Backdrop = styled.TouchableWithoutFeedback`
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
`;

const ModalBackground = styled.View`
  border-top-left-radius: ${props => props.customTheme.borderRadius};
  border-top-right-radius: ${props => props.customTheme.borderRadius};
  overflow: hidden;
  padding: ${props => props.customTheme.padding};
  box-shadow: 0px 2px 7px rgba(0,0,0,.1);
  elevation: 1;
  margin-top: auto;
  background-color: ${({ customTheme, theme }) => customTheme.isTransparent ? 'transparent' : theme.colors.basic050};
  margin-horizontal: ${({ sideMargins }) => sideMargins || 0}px;
`;

const getModalContentPadding = (showHeader: boolean) => {
  if (showHeader) {
    return '0';
  }
  return `${spacing.rhythm}px 0 0`;
};

const ModalContent = styled.View`
  flex-direction: column;
  ${({ fullScreen, showHeader }) => fullScreen && showHeader && `
    padding: ${getModalContentPadding(showHeader)};
  `}
  ${({ fullScreen }) => fullScreen && `
    flex: 1;
  `}
`;

const ModalOverflow = styled.View`
  width: 100%;
  height: 100px;
  margin-bottom: -100px;
  background-color: ${({ theme }) => theme.colors.basic050};
`;

class SlideModal extends React.Component<Props, State> {
  _modalRef = React.createRef<Modal>();

  state = {
    contentHeight: 0,
  };

  close: () => void = () => {
    if (this._modalRef.current) this._modalRef.current.close();
  }

  onModalBoxLayout = (event: LayoutEvent) => {
    const height = event.nativeEvent?.layout?.height || 0;
    if (this.state.contentHeight !== height) {
      this.setState({
        contentHeight: height,
      });
    }
  }

  render() {
    const {
      children,
      title,
      noClose,
      fullScreen,
      showHeader,
      hideHeader,
      centerTitle,
      backgroundColor: bgColor,
      eventDetail,
      noSwipeToDismiss,
      theme,
      noPadding,
      headerLeftItems,
      sideMargins,
      noTopPadding,
      headerProps = {},
      insetTop,
      centerFloatingItem,
    } = this.props;

    const customTheme = getTheme(this.props);
    const colors = getThemeColors(theme);
    const backgroundColor = bgColor || colors.basic070;

    const showModalHeader = ((!fullScreen || showHeader) && !hideHeader) || !isEmpty(headerProps);
    let leftItems = [];
    const centerItems = centerTitle ? [{ title }] : [];
    const rightItems = [{
      close: !noClose,
    }];
    if (!centerTitle) {
      leftItems.push({ title });
    }
    if (headerLeftItems) {
      leftItems = [...leftItems, ...headerLeftItems];
    }

    const modalInner = (
      <React.Fragment>
        {showModalHeader &&
          // $FlowFixMe: flow update to 0.122
          <HeaderBlock
            leftItems={leftItems}
            centerItems={centerItems}
            rightItems={rightItems}
            noBottomBorder
            noPaddingTop
            onClose={this.close}
            wrapperStyle={{ backgroundColor: 'transparent' }}
            noHorizontalPadding={!fullScreen && !noPadding}
            leftSideFlex={centerTitle ? undefined : 4}
            noBack
            forceInsetTop={insetTop ? 'always' : 'never'} // eslint-disable-line i18next/no-literal-string
            {...headerProps}
          />
        }
        <ModalContent
          fullScreen={fullScreen}
          showHeader={showHeader}
        >
          {children}
        </ModalContent>
        <ModalOverflow />
      </React.Fragment>
    );


    const modalContent = () => {
      if (fullScreen) {
        return (
          <Wrapper onLayout={this.onModalBoxLayout} fullScreen>
            {modalInner}
          </Wrapper>
        );
      }

      if (eventDetail) {
        return (
          <ModalBackground onLayout={this.onModalBoxLayout} customTheme={customTheme} sideMargins={sideMargins}>
            { children }
          </ModalBackground>
        );
      }

      return (
        <ModalBackground onLayout={this.onModalBoxLayout} customTheme={customTheme} sideMargins={sideMargins}>
          { modalInner }
        </ModalBackground>
      );
    };

    const animationTiming = 400;

    /* eslint-disable i18next/no-literal-string */
    const fwdProps: ModalProps = pick(
      this.props,
      'onModalWillHide',
      'onModalHide',
      'onModalShow',
      'scrollOffset',
      'scrollOffsetMax',
      'scrollTo',
      'avoidKeyboard',
    );
    /* eslint-enable i18next/no-literal-string */

    return (
      <Modal
        ref={this._modalRef}
        backdropOpacity={fullScreen ? 1 : 0.7}
        backdropColor={fullScreen ? backgroundColor : '#000000'}
        animationInTiming={animationTiming}
        animationOutTiming={animationTiming}
        swipeDirection={noSwipeToDismiss ? undefined : 'down'}
        style={{
          margin: 0,
          position: 'relative',
          zIndex: 10,
        }}
        {...fwdProps}
      >
        <ContentWrapper fullScreen={fullScreen} bgColor={backgroundColor} noTopPadding={noTopPadding}>
          {!fullScreen && (
            <Backdrop onPress={this.close}>
              <ContentWrapper />
            </Backdrop>
          )}
          {!!centerFloatingItem && (
            <Wrapper
              style={{
                elevation: 2,
                zIndex: 11,
                marginTop: -1 * this.state.contentHeight,
                marginBottom: 0,
              }}
            >
              {centerFloatingItem}
            </Wrapper>
          )}
          {modalContent()}
        </ContentWrapper>
      </Modal>
    );
  }
}

export type SlideModalInstance = SlideModal;

const ThemedSlideModal: React.AbstractComponent<OwnProps, SlideModal> = withTheme(SlideModal);
export default ThemedSlideModal;
