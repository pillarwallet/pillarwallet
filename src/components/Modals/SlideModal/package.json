{
  "name": "SlideModal.js",
  "version": "0.0.0",
  "main": "./SlideModal.js"
}
