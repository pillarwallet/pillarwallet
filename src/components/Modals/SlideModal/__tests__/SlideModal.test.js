// @flow
/*
    Pillar Wallet: the personal data locker
    Copyright (C) 2019 Stiftung Pillar Project

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
*/
import * as React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BaseText } from 'components/Typography';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'utils/themes';
import configureStore from '../../../../configureStore';
import SlideModal from '../SlideModal';

const { store } = configureStore();

describe('Slide Modal', () => {
  it('should render SlideModal correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <ThemeProvider theme={defaultTheme}>
          <SlideModal title="title" />
        </ThemeProvider>
      </Provider>).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render SlideModal with content', () => {
    const ChildContent = () => <BaseText>Test</BaseText>;
    const wrapper = shallow(
      <ThemeProvider theme={defaultTheme}>
        <SlideModal title="title">
          <ChildContent />
        </SlideModal>
      </ThemeProvider>);
    expect(wrapper.find(ChildContent)).toHaveLength(1);
  });
});
