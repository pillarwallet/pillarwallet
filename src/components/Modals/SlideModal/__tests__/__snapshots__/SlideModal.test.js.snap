// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide Modal should render SlideModal correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  onSwipeComplete={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection="down"
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onSwipeComplete={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={[Function]}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "margin": 0,
        "position": "relative",
        "top": 0,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
        "zIndex": 10,
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="down"
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        bgColor="#FAFAFA"
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "height": "100%",
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <View
          customTheme={
            Object {
              "borderRadius": "30px",
              "padding": "0 16px",
            }
          }
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "borderTopLeftRadius": 30,
                "borderTopRightRadius": 30,
                "elevation": 1,
                "marginHorizontal": 0,
                "marginTop": "auto",
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 0,
                "shadowColor": "rgba(0,0,0,.1)",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 7,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "transparent",
                "width": "100%",
              }
            }
          >
            <View
              androidStatusbarHeight={42}
              noPaddingTop={true}
              onLayout={[Function]}
              pointerEvents="box-none"
              style={
                Object {
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                }
              }
            >
              <View
                noHorizonatalPadding={true}
                style={
                  Array [
                    Object {
                      "minHeight": 58,
                      "paddingVertical": 15,
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "width": "100%",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    sideFlex={4}
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 4,
                          "flexShrink": 1,
                          "flexWrap": "wrap",
                          "justifyContent": "flex-start",
                          "minHeight": 28,
                        },
                        Object {},
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                          "flexWrap": "wrap",
                        }
                      }
                    >
                      <Text
                        centerText={false}
                        style={
                          Array [
                            Object {
                              "color": "#0A1427",
                              "fontFamily": "EuclidCircularB-Regular",
                              "textAlignVertical": "center",
                            },
                            Array [
                              Object {
                                "color": "#0A1427",
                                "fontFamily": "EuclidCircularB-Medium",
                                "textAlignVertical": "center",
                              },
                              Array [
                                Object {
                                  "color": "#0A1427",
                                  "fontSize": 16,
                                  "lineHeight": 26,
                                  "textAlign": "left",
                                },
                                Object {},
                              ],
                            ],
                          ]
                        }
                      >
                        title
                      </Text>
                    </View>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "flexWrap": "wrap",
                          "justifyContent": "flex-end",
                          "minHeight": 28,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "marginBottom": -20,
                          "marginLeft": -12,
                          "marginRight": -20,
                          "marginTop": -20,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "flex-end",
                            "alignSelf": "center",
                            "justifyContent": "center",
                            "opacity": 1,
                            "paddingBottom": 20,
                            "paddingLeft": 20,
                            "paddingRight": 20,
                            "paddingTop": 20,
                            "position": "relative",
                          }
                        }
                      >
                        <View
                          name="close"
                          style={
                            Object {
                              "color": "#0A1427",
                              "fontSize": 14,
                              "marginBottom": undefined,
                              "marginLeft": undefined,
                              "marginRight": undefined,
                              "marginTop": undefined,
                              "paddingTop": 0,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 100,
                  "marginBottom": -100,
                  "width": "100%",
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <View
        androidStatusbarHeight={42}
        opacity={0}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "left": 0,
              "marginTop": 42,
              "opacity": 0,
              "paddingBottom": 40,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 40,
              "position": "absolute",
              "top": 0,
              "width": "100%",
              "zIndex": 1000,
            },
            Object {
              "transform": Array [
                Object {
                  "translateY": -260,
                },
              ],
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "opacity": 1,
                  "position": "relative",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": 1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              accessible={true}
              borderRadius={20}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              opacity={1}
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 20,
                    "flexDirection": "column",
                    "justifyContent": "flex-start",
                    "opacity": 1,
                    "width": "100%",
                  },
                  Object {},
                ]
              }
            >
              <View
                onLayout={[Function]}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "flex-start",
                        "flexDirection": "row",
                        "paddingBottom": 14,
                        "paddingLeft": 20,
                        "paddingRight": 55,
                        "paddingTop": 14,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#000",
                        "fontSize": 16,
                      }
                    }
                  >
                    👌
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "height": 0,
                          "width": 18,
                        },
                        undefined,
                      ]
                    }
                    w={18}
                  />
                  <Text
                    regular={true}
                    style={
                      Array [
                        Object {
                          "color": "#0A1427",
                          "fontFamily": "EuclidCircularB-Regular",
                          "fontSize": 14,
                          "lineHeight": 22,
                          "textAlignVertical": "center",
                        },
                        Array [
                          Object {
                            "color": "#0A1427",
                            "fontFamily": "EuclidCircularB-Medium",
                            "fontSize": 14,
                            "lineHeight": 22,
                            "textAlignVertical": "center",
                          },
                          Object {
                            "flex": 1,
                          },
                        ],
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                      "paddingRight": 10,
                      "paddingTop": 10,
                      "position": "absolute",
                      "right": 8,
                      "top": 7,
                    }
                  }
                >
                  <View
                    name="rounded-close"
                    style={
                      Array [
                        Object {
                          "color": "#ededed",
                          "fontSize": 16,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.4)",
              "flex": 1,
              "justifyContent": "flex-end",
              "opacity": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "elevation": 4,
                "maxHeight": 667,
                "minHeight": 56,
                "opacity": 1,
                "padding": 15,
              }
            }
          >
            <RCTScrollView
              data={Array []}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              style={
                Array [
                  Object {
                    "marginHorizontal": -15,
                  },
                  Object {
                    "marginTop": 0,
                  },
                ]
              }
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View />
            </RCTScrollView>
          </View>
        </View>
      </Modal>
    </View>
  </View>
</Modal>
`;
